### ISSUE TITLE
Incorrect evaluation at objects that are not numbers

### ISSUE THREAD

**Issue by cafaxo** [JuliaMath/Polynomials.jl#189]
Suppose we have a polynomial `p = a_0 x^0 + a_1 x^1 + ... + a_n x^n` where the a_i are in a ring **R**. Then I would expect `p(α) = a_0 α^0 + a_1 α^1 + ... + a_n α^n` to hold for all α in a ring **S** where **R** is a subring of **S**.

This is currently not the case. Example:
```julia
julia> Poly([0,0,1])([0 1; 0 0])
2×2 Array{Int64,2}:
 0  1
 0  0
```
but [0 1; 0 0]^2 == [0 0; 0 0].

**Comment by jverzani** [comment]
This is due to this definition `polyval(p::Poly, v::AbstractArray) = map(x->polyval(p, x), v)` which is basically doing `p.(A)` when `A` is a matrix. In #179 this call is an error. I wonder if it makes better sense to change the definition for `p(A)` to what you are expecting, as that seems natural. In #179 it seems to only require changing a `+` to a `.+`.

**Comment by jverzani** [comment]
Closed by #196
